### ISSUE TITLE
The semicolon `declaration-block-trailing-semicolon` writes is formatted by neither rule that formats the others

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#354]
Stylelint runs the rules of a plugin in the order the configuration spells them, and it runs each of them once. The semicolon `declaration-block-trailing-semicolon` writes is therefore never seen by the two rules that format the semicolons already in the file, whenever those rules are listed first.

## The run

Fixture `@media screen{\na{b:c;d:e}\n}\n`, both rules at once, only the order changed:

```
declaration-block-semicolon-newline-before first -> "@media screen{\na{b:c\n;d:e;}\n}\n"     one warning left
declaration-block-trailing-semicolon first       -> "@media screen{\na{b:c\n;d:e\n;}\n}\n"   nothing left to report
```

The semicolon of the first declaration is moved onto a line of its own; the one written at the end of the block is not, and the block ends up spelling its two semicolons two ways. The same happens with a space instead of a break:

```
declaration-block-semicolon-space-before first -> "@media screen{\na{b:c ;d:e;}\n}\n"    one warning left
declaration-block-trailing-semicolon first     -> "@media screen{\na{b:c ;d:e ;}\n}\n"   nothing left to report
```

A second run of `--fix` repairs both, and the first run reports nothing at all.

## Why

`lintPostcssResult.mjs` of `stylelint@17.14.1` sorts the rules of a run by their place in Stylelint's own registry, and a plugin's rules are not in it:

```js
const rulesOrder = new Map(Object.keys(rules).map((ruleName, index) => [ruleName, index]))

ruleNames = Object.keys(config.rules).sort((a, b) => (rulesOrder.get(a) ?? -1) - (rulesOrder.get(b) ?? -1))
```

Every rule of this plugin resolves to `-1`, the sort is stable, and so the order is the one the user's configuration happens to spell. `lintPostcssResult` is called once per lint; there is no loop to a fixed point.

## The counterparts

- `declaration-block-semicolon-newline-before`
- `declaration-block-semicolon-space-before`

The default configuration of `@stylistic/stylelint-config` escapes this today by accident rather than by design: it sets `declaration-block-semicolon-space-before` to `never`, and a semicolon written tight against the value is what `never` asks for anyway. Under `always` the hole opens.

## What ends it

`declaration-block-trailing-semicolon` has to write the semicolon in its finished form — with whatever stands in front of it that the configuration asks for — rather than a bare character for another rule to correct afterwards. That means reading the settings of the two rules above out of `result.stylelint.config`, which holds every rule's normalised settings and is assigned before any rule runs, through a shared helper.

The measurement ran 79 rules under 208 primary options over eight fixtures of plain CSS.

One of the four classes of [#356](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/356), which is about the class rather than about any member of it.
